app/controllers/students_controller.rb

```diff
@@ -37,13 +41,18 @@ def update
   end
 
   def destroy
-    @student = Student.find(params[:id])
+    authorize_user!
+    @student = current_user.students.find(params[:id])
     @student.destroy
     redirect_to students_path
   end
 
   private
 
+  def authorize_user!
+    raise Pundit::NotAuthorizedError if current_user.admin?
+  end
```

Comment:
this works nicely.. we can move it to somewhere re-usable soon, so we are not repeating ourselves